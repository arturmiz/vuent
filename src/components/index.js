import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
