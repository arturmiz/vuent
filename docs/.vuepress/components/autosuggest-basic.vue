<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
