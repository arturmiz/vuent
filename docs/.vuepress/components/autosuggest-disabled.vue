<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
