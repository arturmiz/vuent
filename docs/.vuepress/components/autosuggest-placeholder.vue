<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
