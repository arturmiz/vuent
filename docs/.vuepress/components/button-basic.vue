<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
