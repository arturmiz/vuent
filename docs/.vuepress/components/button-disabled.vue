<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
