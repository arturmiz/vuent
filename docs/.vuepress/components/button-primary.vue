<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
