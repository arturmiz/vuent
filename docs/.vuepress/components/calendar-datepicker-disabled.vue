<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
