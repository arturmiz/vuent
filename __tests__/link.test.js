import { createLocalVue, mount } from '@vue/test-utils';
import { VntLink } from '@/components';
import { isInstalled } from './utils';

describe('Link', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntLink);

    expect(isInstalled(localVue, VntLink)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntLink);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('target is #', () => {
      expect(wrapper.vm.href).toBe('#');
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('click handler is undefined', () => {
      expect(wrapper.vm.click).toBeUndefined();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntLink, {
        propsData: {
          disabled: true
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

  });

  describe('can set target', () => {

    beforeAll(() => {
      wrapper = mount(VntLink, {
        propsData: {
          href: 'http://fluent.microsoft.com'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('is correct', () => {
      expect(wrapper.vm.href).toBe('http://fluent.microsoft.com');
    });

  });

  describe('can create proper link', () => {

    beforeAll(() => {
      wrapper = mount(VntLink, {
        propsData: {
          href: 'http://fluent.microsoft.com'
        },
        slots: {
          default: 'Fluent'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('click handler is not invoked', () => {
      const mockClick = jest.fn();
      wrapper.vm.handlerClick = mockClick;
      wrapper.find('a').trigger('click');
      expect(mockClick).not.toBeCalled();
    });

  });

  describe('when passed other standard href attributes', () => {

    test('passes correctly target attr', () => {
      wrapper = mount(VntLink, {
        propsData: {
          href: 'http://fluent.microsoft.com'
        },
        attrs: {
          target: '_blank'
        }
      });

      // TODO: replace with following once https://github.com/vuejs/vue-test-utils/issues/146 is fixed
      // expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.vm.$attrs.target).toBe('_blank');
    });

  });

  test('when clicked invokes passed handler', () => {
    const mockClick = jest.fn();
    wrapper = mount(VntLink, {
      propsData: {
        click: mockClick
      }
    });
    wrapper.find('a').trigger('click');
    expect(mockClick).toBeCalled();
  });

  describe('when both href and click handler passed', () => {

    test('invokes the click handler', () => {
      const mockClick = jest.fn();
      wrapper = mount(VntLink, {
        propsData: {
          click: mockClick,
          href: 'http://fluent.microsoft.com'
        }
      });
      wrapper.find('a').trigger('click');
      expect(mockClick).toBeCalled();
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

});

