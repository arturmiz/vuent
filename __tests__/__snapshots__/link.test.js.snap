// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link by default renders correctly 1`] = `<a href="#" class="vnt-link">hyperlink</a>`;

exports[`Link can be disabled renders correctly 1`] = `<span class="vnt-link vnt-link--disabled">hyperlink</span>`;

exports[`Link can create proper link renders correctly 1`] = `<a href="http://fluent.microsoft.com" class="vnt-link">Fluent</a>`;

exports[`Link can set target renders correctly 1`] = `<a href="http://fluent.microsoft.com" class="vnt-link">hyperlink</a>`;

exports[`Link when both href and click handler passed renders correctly 1`] = `<a href="http://fluent.microsoft.com" class="vnt-link">hyperlink</a>`;
